src/Microsoft.SqlTools.ServiceLayer/Connection/ConnectionInfo.cs

```diff
@@ -99,6 +99,15 @@ public ICollection<string> AllConnectionTypes
             }
         }
 
+        public bool HasConnectionType(string connectionType)
+        {
+            if (connectionType == null)
+            {
+                connectionType = ConnectionType.Default;
+            }
+            return ConnectionTypeToConnectionMap.ContainsKey(connectionType);

```

Comment:
And then it's all nicely atomic and you don't need to worry about race conditions.
BUT, if ConnectionTypeToConnectionMap isn't a ConcurrentDictionary, I'll concede on this request for now.